Theming

RazCoH opened the pull request:
Implement custom UI theming for colors, shapes and typography. 
Create easy separation between dark and light modes.

Commit:
Update dependencies

Commit:
implement color and typography custom theme

Commit:
Merge branch 'master' into theming

# Conflicts:
#	app/src/main/java/com/example/composetemplate/ui/theme/Color.kt
#	gradle/libs.versions.toml

Commit:
no message

Commit:
Merge branch 'master' into theming

Commit:
Implement the custom theme in the project

ShonOhana commented on app/build.gradle.kts:
```diff
@@ -117,7 +117,4 @@ dependencies {
     implementation(libs.androidx.datastore.preferences.core)
-
-    //Accompanist
-    implementation(libs.accompanist.systemuicontroller)
```
what is the soulution instead of it?

ShonOhana commented on app/src/main/java/com/example/composetemplate/presentation/common/FullScreenProgressBar.kt:
```diff
@@ -21,7 +22,7 @@ fun FullScreenProgressBar(isLoading: Boolean) {
         ) {
             CircularProgressIndicator(
-                color = Color.White, // Set color as needed
+                color = CustomTheme.colors.text,
```
why color named text on indicator?

RazCoH replied:
Ok I will change it

ShonOhana commented on app/src/main/java/com/example/composetemplate/presentation/common/LoginTextField.kt:
```diff
@@ -49,24 +47,24 @@ fun LoginTextField(
                         if (!isValid) Color.Red else Color.Transparent
                     }
-                ), CircleShape
+                ), CustomTheme.shapes.textField
```
is textField the right name for circle shape? what if there are textfield that dont need circle shape?

RazCoH replied:
That's the preferred shape for the text field in the app. It represents the text field shape, not a circular shape. If the text field becomes rectangular in the future, I will update the shape accordingly.

RazCoH replied:
I will change to circleTextFiled

ShonOhana commented on app/src/main/java/com/example/composetemplate/presentation/common/LoginTextField.kt:
```diff
@@ -49,24 +47,24 @@ fun LoginTextField(
+        shape = CustomTheme.shapes.textField,
         colors = TextFieldDefaults.colors(
-            cursorColor = if (isSystemInDarkTheme()) Color.White else Color.Black,
+            cursorColor = CustomTheme.colors.cursor,
```
is that know automaticaly id its dark theme or no? I dont remember

ShonOhana commented on app/src/main/java/com/example/composetemplate/presentation/common/LoginTextField.kt:
```diff
@@ -49,24 +47,24 @@ fun LoginTextField(
-            focusedContainerColor = Color.DarkGray,
-            focusedTextColor = LoginColorEnable,
-            unfocusedTextColor = if (isValid) LoginColorEnable else Color.Red
+            unfocusedContainerColor = CustomTheme.colors.editTextContainer,
```
same is here, this is edittext of login, maybe naming can change. editTextContainer can be for other edit text in app

Commit:
Pr rejects , fix naming

Commit:
Clear status bar color
I exported the code that paints the status bar to composable function PaintStatusBar

Commit:
Update dependancies
